protocol/client.go

```diff
@@ -0,0 +1,203 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto/sign"
+	"github.com/coniks-sys/coniks-go/crypto/vrf"
+	m "github.com/coniks-sys/coniks-go/merkletree"
+)
+
+type ProofType int
+
+const (
+	InvalidProof ProofType = iota
+	ProofOfAbsence
+	ProofOfInclusion
+)
+
+// ConiksClient stores the current state of a CONIKS client.
+// This includes the current epoch and SignedTreeRoot value,
+// as well as directory's configurations
+// (e.g., wherether the TemporaryBinding extension is being used or not).
+// This struct also stores the verification results including
+// the type of returned proof.
+// The client should create a new ConiksClient instance only once,
+// when she registers her binding with a ConiksDirectory.
+// This ConiksClient instance then will be used to verify
+// the returned response from a ConiksDirectory.
+type ConiksClient struct {
+	CurrentEpoch uint64
+	SavedSTR     []byte
+
+	// verification results
```

Comment:
If the proof type check is done in this package, do we still need these two fields?